src/test/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/nodes/UserIDCauseRestrictionPipelineTest.java

```diff
@@ -0,0 +1,125 @@
+        final WorkflowJob project = j.jenkins.createProject(WorkflowJob.class, "pipeline_demo");
+        project.setDefinition(new CpsFlowDefinition(
+                "node('" + slave.getNodeName() + "') {\n" +
+                "    sh('echo hello')\n" +
+                "}", true));
+      
+        // schedule a build for the pipeline job as the user who is accepted by the test slave
+        WorkflowRun run = runAsUserAndWaitForCompletion(project, TEST_USERNAME);
+        assertThat("Job restriction should allow pipeline jobs", run.getResult(), equalTo(Result.SUCCESS));
+    }
+    
+    @Test
+    public void nodeShouldNotAllowPipelineRunsByUnacceptedUser_UserIdCause() throws Exception {
+        DumbSlave slave = j.createOnlineSlave();
+        List<UserSelector> listUserSelector = new ArrayList<UserSelector>(1);
+        listUserSelector.add(new UserSelector(TEST_USERNAME_2));
+      
+        // add StartedByUser Restriction property to node
+        slave.getNodeProperties().add(new JobRestrictionProperty(new StartedByUserRestriction(listUserSelector, true, true, false)));
+      
+        final WorkflowJob project = j.jenkins.createProject(WorkflowJob.class, "pipeline_demo");
+        project.setDefinition(new CpsFlowDefinition(
+                "node('" + slave.getNodeName() + "') {\n" +
+                "    sh('echo hello')\n" +
+                "}", true));
+      
+        // schedule a build for the pipeline job as the user who is not accepted by the test slave
+        QueueTaskFuture<WorkflowRun> runFuture = runAsUser(project, TEST_USERNAME);
+        WorkflowRun run = runFuture.waitForStart();
+        Thread.sleep(2000); // Any way to do it better
+        List<StepExecution> executions = run.getExecutionPromise().get().getCurrentExecutions(true).get();
+        assertThat("Missing current head", executions.size(), equalTo(1));
+        StepExecution currentExecution = executions.get(0);
+        assertThat(currentExecution, instanceOf(ExecutorStepExecution.class));
+        ExecutorStepExecution exec = ((ExecutorStepExecution)currentExecution);
+
+        // Now we know there is a waiting execution
+        List<Queue.BuildableItem> buildableItems = j.jenkins.getQueue().getBuildableItems();
+        assertThat("There is no buildable items in the queue. The job is not blocked", buildableItems.size(), equalTo(1));
+        assertThat(buildableItems.get(0).task, instanceOf(ExecutorStepExecution.PlaceholderTask.class));
+        assertThat("The task is not blocked", buildableItems.get(0).getCauseOfBlockage(), not(nullValue()));
+    }
+
+    @Nonnull
+    private QueueTaskFuture<WorkflowRun> runAsUser(WorkflowJob job, String username) throws InterruptedException, ExecutionException {
+        QueueTaskFuture<WorkflowRun> runFuture;
+        SecurityContext old = ACL.impersonate((User.getById(username, true)).impersonate());
+        try {
+            runFuture = job.scheduleBuild2(0, new CauseAction(new Cause.UserIdCause()));
```

Comment:
I suppose this was the impetus for https://github.com/jenkinsci/jenkins/pull/3162?

Reply:
yes